internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -298,11 +384,34 @@ func convertVolumes(input map[string]manifest.Volume) ([]*template.Volume, error
 	return output, nil
 }
 
-func convertEFSConfiguration(in *manifest.EFSVolumeConfiguration) (*template.EFSVolumeConfiguration, error) {
+func convertEFS(in *manifest.EFSConfigOrID) (*template.EFSVolumeConfiguration, error) {
 	// If there is no EFS information, just add the Name to the volume.
 	if in == nil {
 		return nil, nil
 	}
+	// UID and GID should not be specified for non-managed volumes.
+	if !in.UseManagedFS() && !in.Config.IsEmpty() {
```

Comment:
do we need to guard again against `in.UseManagedFS()`? Maybe we can make assumptions here saying the caller should ensure to call this function for non-managed FSs